clang/test/SemaSYCL/implicit_kernel_type.cpp

```diff
@@ -50,34 +19,70 @@ class myWrapper2;
 
 int main() {
   queue q;
-#ifndef __SYCL_UNNAMED_LAMBDA__
-  // expected-note@+3 {{InvalidKernelName1 declared here}}
-  // expected-note@+4{{in instantiation of function template specialization}}
-  // expected-error@28 {{kernel needs to have a globally-visible name}}
+
+#if defined(WARN)
+  // expected-warning@Inputs/sycl.hpp:211 {{Passing of kernel functions by reference is a SYCL 2020 extension}}
+  // expected-error@Inputs/sycl.hpp:220 {{kernel needs to have a globally-visible name}}
+  // expected-note@+8 {{InvalidKernelName1 declared here}}
+#elif defined(ERROR)
+  // expected-error@Inputs/sycl.hpp:211 {{Passing of kernel functions by reference is a SYCL 2020 extension}}
+  // expected-error@Inputs/sycl.hpp:220 {{kernel needs to have a globally-visible name}}
+  // expected-note@+4 {{InvalidKernelName1 declared here}}
+#elif defined(__SYCL_UNNAMED_LAMBDA__)
+  // expected-error@Inputs/sycl.hpp:211 {{Passing of kernel functions by reference is a SYCL 2020 extension}}
```

Comment:
Since the test is not about the difference between ways of passing kernel functions in different sycl standards, can we suppress this error/warning by passing `-sycl-std=2020` like you did in the other test?